staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -582,6 +582,12 @@ func (c *Controller) needsUpdate(oldService *v1.Service, newService *v1.Service)
 		return true
 	}
 
+	if len(oldService.Spec.IPFamilies) != len(newService.Spec.IPFamilies) {
```

Comment:
Does the order matter in this case? Should we check order in addition to length of ip families?

Reply:
see comment https://github.com/kubernetes/kubernetes/pull/116520#issuecomment-1467232074

Reply:
can we add a comment in the code with this context?

Reply:
can we add a comment in the code with this context?

Reply:
done, add comment with this context. PTAL